strawberryfields/ops.py

```diff
@@ -396,7 +397,7 @@ def __or__(self, reg):
         if (not reg) or (self.ns is not None and self.ns != len(reg)):
             raise ValueError("Wrong number of subsystems.")
         # append it to the Program
-        reg = Program._current_context.append(self, reg)
+        reg = pu.Program_current_context.append(self, reg)
```

Comment:
program.py depends on devicespecs, which depends on gbs.py, which depends on ops.py.
If ops.py imports Program from program.py we get a circular import.

I thought the easiest way to solve it was to move the `_current_context` variable to another module. It's not as neat as having it as a class attribute, but it works.

Another way to solve circular imports would be to use only absolute imports, but it would be a larger change.